### ISSUE TITLE
Tests for twisted.web should conform to the new style of naming test classes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7689]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)| @eeshangarg reported|
|-|-|
|Trac ID|trac#7689|
|Type|enhancement|
|Created|2014-10-18 13:44:33Z|
|Branch|https://github.com/twisted/twisted/tree/web-tests-naming-7689|

This ticket is a follow-up to ticket [[ticket:7032]].

Attachments:

* [bug-7689.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d80/d80bc4c2dfdcecc2d6ae22d25e4feba216f991cf/268df1ebb2e53967d82dd2d892800ae2220bbd11.patch) (18801 bytes) - added by eeshangarg on 2014-10-18 14:04:43Z - 
* [bug-7689_2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d80/d80bc4c2dfdcecc2d6ae22d25e4feba216f991cf/aeeb587d46da7299a64295280695e7f8988c278c.patch) (17481 bytes) - added by eeshangarg on 2014-10-21 13:03:55Z - 
* [bug-7689_3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d80/d80bc4c2dfdcecc2d6ae22d25e4feba216f991cf/f53b9ae85fab76f04f8743c3edcbfc909aad281c.patch) (17587 bytes) - added by eeshangarg on 2014-11-19 11:54:56Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7689 7689
type__enhancement enhancement
reporter__eeshangarg eeshangarg
priority__normal normal
milestone__ 
branch__branches_web_tests_naming_7689 branches/web-tests-naming-7689
branch_author__eeshangarg eeshangarg
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1413639873868082 1413639873868082
changetime__1417019894796448 1417019894796448
version__None None
owner__exarkun exarkun
cc__eeshangarg cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)<a name="note_2"></a>|@eeshangarg set owner to @glyph|
|-|-|

Hi! I just added a patch that changes all of the tests in twisted.web to use the new style of naming test classes as discussed in ticket [ticket:7032].

Thanks!
Eeshan

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @eeshangarg|
|-|-|

Thanks for your efforts here.

For the record, I think it's a bad idea to make vast changes in some narrow scope.  Instead, I prefer actual maintenance of specific areas of the code base.  It yields better results overall as well as less churn.

Nevertheless, here is a review of this patch.

  1. The renaming of `twisted.web.test._util.FlattenTestCase` is incorrect.  Its original name is correct.  It is not a test suite.  It is a specialized `TestCase` subclass.
  1. As long as `XMLRPCTestCase2` is being renamed, it should probably be given a name that identifies its purpose with more detail than "2" does.

**Comment by twisted-trac** [comment]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)<a name="note_4"></a>|@eeshangarg set owner to @exarkun|
|-|-|

Hello Mr. exarkun

I addressed the two concerns that you mentioned in your review of my first patch in my second patch called bug-7689_2.patch​. If I missed something, or if there is something that could be improved, please let me know and I will make the required changes as soon as possible. I am having a lot of fun contributing and want to keep going, and would love to close this ticket. :-)

Thanks a lot for the review!

Regards,
Eeshan Garg

**Comment by twisted-trac** [comment]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)<a name="note_5"></a>|@eeshangarg commented|
|-|-|

Hi!

I just submitted a new patch for this called bugfix_7689_3.patch, which includes a .misc news file along with the contents of the previous patch. :)

Thanks & regards,
Eeshan Garg

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [43496]) Branching to 'web-tests-naming-7689'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [43497]) Apply bug-7689_3.patch 

Refs [#7689](https://github.com/twisted/twisted/issues/7689)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

Thanks.  These changes all look fine to me.  Buildbot, apart from twistedchecker, seems happy (and twistedchecker is just confused by the new names).  I'll merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [43500]) Merge web-tests-naming-7689

Author: eeshangarg
Reviewer: exarkun
Fixes: [#7689](https://github.com/twisted/twisted/issues/7689)

Rename all of the test case classes in Twisted Web to conform to the
recently documented test case class naming convention.
